fgrpc/pingsrv.go

```diff
@@ -110,16 +111,24 @@ func PingServerTCP(port, cert, key, healthServiceName string, maxConcurrentStrea
 
 // PingClientCall calls the ping service (presumably running as PingServer on
 // the destination). returns the average round trip in seconds.
-func PingClientCall(serverAddr string, n int, payload string, delay time.Duration, tlsOpts *fhttp.TLSOptions) (float64, error) {
+func PingClientCall(serverAddr string, n int, payload string, delay time.Duration, tlsOpts *fhttp.TLSOptions, md metadata.MD) (float64, error) {
 	o := GRPCRunnerOptions{Destination: serverAddr, TLSOptions: *tlsOpts}
 	conn, err := Dial(&o) // somehow this never seem to error out, error comes later
 	if err != nil {
 		return -1, err // error already logged
 	}
 	msg := &PingMessage{Payload: payload, DelayNanos: delay.Nanoseconds()}
 	cli := NewPingServerClient(conn)
+	mkCtx := func() context.Context {
+		return context.Background()
+	}
+	if md.Len() != 0 {
+		mkCtx = func() context.Context {
+			return metadata.NewOutgoingContext(context.Background(), md)
```

Comment:
this (as the linters complain - btw do run `make lint` (or `make local-lint` if you have same version of go and `golangci-lint` on your local machine) is a bit odd

given all this is in same context/loop - do like you did above and just have 1 ctx variable?

Reply:
I downloaded `golangci-lint` and did some checking and fixed it.